external/js/cothority/src/byzcoin/byzcoin-rpc.ts

```diff
@@ -284,9 +301,49 @@ export default class ByzCoinRPC implements ICounterUpdater {
         });
 
         const rep = await this.conn.send<GetSignerCountersResponse>(req, GetSignerCountersResponse);
+        const counters = this.counters();
+        ids.forEach((counter, i) => counters.set(counter.toString(), rep.counters[i]));
         return rep.counters.map((c) => c.add(add));
     }
 
+    /**
+     * This keeps track of the know counters, thus speeding up signing with known keys. It
+     * also solves the problem of counters sometimes being out of synch with the chain.
+     * After the call returns, all signers are in ClientTransaction.counters.
+     *
+     * @param rpc should allow updates for counters
+     * @param signers the signers needed
+     */
+    async updateCachedCounters(signers: IIdentity[]): Promise<Long[]> {
+        const counters = this.counters();
+        const newSigners = signers.filter((signer) => {
+            return counters.has(signer.toString()) === false;
```

Comment:
IMHO it's much clearer without the `=== false` and with a `!`. Ignore if you want, not a big deal.

Reply:
I can't do that anymore - @tharvik is watching over me and he's making sure that everything is correctly typed!

Reply:
I agree with @Gilthoniel on this one, comparison with bool are redundant, but when an object get casted to bool (to test for undef really), it is type-wise wrong.
```suggestion
            return !counters.has(signer.toString());
```